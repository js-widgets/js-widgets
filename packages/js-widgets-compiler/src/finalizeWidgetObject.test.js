const finalizeWidgetObject = require('./finalizeWidgetObject');
jest.mock('./util/currentTime', () => () => '2021-01-04T11:17:22.940Z');

describe('finalizeWidgetObject', () => {
  it('should fill in the details', () => {
    expect.assertions(2);
    expect(
      finalizeWidgetObject(
        { shortcode: 'foo' },
        `${__dirname}/../tests/exampleDir1`,
        `${__dirname}/../tests/exampleDir1`,
      ),
    ).toMatchSnapshot();
    expect(
      finalizeWidgetObject(
        { shortcode: 'foo', createdAt: 'a made up date' },
        `${__dirname}/../tests/exampleDir1`,
        `${__dirname}/../tests/exampleDir1`,
      ),
    ).toMatchSnapshot();
  });
  it('should fail if no widget.json', () => {
    expect.assertions(2);
    expect(() =>
      finalizeWidgetObject(
        { shortcode: 'foo' },
        'tests',
        `${__dirname}/../tests/exampleDir1`,
      ),
    ).toThrow('File widget.json was not found.');
    expect(() =>
      finalizeWidgetObject(
        { shortcode: 'foo' },
        `${__dirname}/../tests/exampleDir1`,
        '/invalid-dir',
      ),
    ).toThrow("ENOENT: no such file or directory, scandir '/invalid-dir'");
  });
  it('should fail if no asset-manifest.json', () => {
    expect.assertions(1);
    expect(() =>
      finalizeWidgetObject(
        { shortcode: 'foo' },
        `${__dirname}/../tests/mocks/fail1`,
        `${__dirname}/../tests/exampleDir1`,
      ),
    ).toThrow('File build/asset-manifest.json was not found.');
  });
});
