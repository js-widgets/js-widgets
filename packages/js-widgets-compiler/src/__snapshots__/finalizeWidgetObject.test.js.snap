// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finalizeWidgetObject should fill in the details 1`] = `
Object {
  "assetManifest": Object {
    "entrypoints": Array [
      "static/js/runtime-main.91bc4a78.js",
      "static/js/2.e7f4434f.chunk.js",
      "static/css/main.036e404b.chunk.css",
      "static/js/main.5939b039.chunk.js",
    ],
    "files": Object {
      "main.css": "/example-widget/static/css/main.036e404b.chunk.css",
      "main.js": "/example-widget/static/js/main.5939b039.chunk.js",
      "precache-manifest.a83582de03f743fdf931b062f9402d7b.js": "/example-widget/precache-manifest.a83582de03f743fdf931b062f9402d7b.js",
      "runtime-main.js": "/example-widget/static/js/runtime-main.91bc4a78.js",
      "static/js/2.e7f4434f.chunk.js": "/example-widget/static/js/2.e7f4434f.chunk.js",
    },
  },
  "createdAt": "2021-01-04T11:17:22.940Z",
  "files": Array [
    "build/asset-manifest.json",
    "exampleDir2/example2.txt",
    "exampleDir2/example3.txt",
    "widget.json",
  ],
  "shortcode": "foo",
  "updatedAt": "2021-01-04T11:17:22.940Z",
}
`;

exports[`finalizeWidgetObject should fill in the details 2`] = `
Object {
  "assetManifest": Object {
    "entrypoints": Array [
      "static/js/runtime-main.91bc4a78.js",
      "static/js/2.e7f4434f.chunk.js",
      "static/css/main.036e404b.chunk.css",
      "static/js/main.5939b039.chunk.js",
    ],
    "files": Object {
      "main.css": "/example-widget/static/css/main.036e404b.chunk.css",
      "main.js": "/example-widget/static/js/main.5939b039.chunk.js",
      "precache-manifest.a83582de03f743fdf931b062f9402d7b.js": "/example-widget/precache-manifest.a83582de03f743fdf931b062f9402d7b.js",
      "runtime-main.js": "/example-widget/static/js/runtime-main.91bc4a78.js",
      "static/js/2.e7f4434f.chunk.js": "/example-widget/static/js/2.e7f4434f.chunk.js",
    },
  },
  "createdAt": "a made up date",
  "files": Array [
    "build/asset-manifest.json",
    "exampleDir2/example2.txt",
    "exampleDir2/example3.txt",
    "widget.json",
  ],
  "shortcode": "foo",
  "updatedAt": "2021-01-04T11:17:22.940Z",
}
`;
